gnomad_constraint/plots/generate_constraint_plots.R

```diff
@@ -84,6 +85,7 @@ constraint_data <- full_join(
 v4 <- filter(constraint_data, .data$v4)
```

Comment:
I also can't add this to the above, I think we should change to handle any restructuring of the input data during loading. 


So, the loading function could be 
```
load_constraint_metrics <- function(
    version = "v4",
    output_basedir = default_output_basedir,
    data_subdir = default_data_subdir,
    downsamplings = FALSE,
    release = TRUE,
    public = TRUE) {
  # Load the constraint metrics data
  # version: The version of the data
  # output_basedir: The base directory for the output
  # data_subdir: The subdirectory for data files
  # downsamplings: Whether to load downsampling data instead of the main data
  # release: Whether to use the release or development version
  # public: Whether to use the public or non-public version
  # Returns: A data frame with constraint metrics data

  full_version <- map_data_version[[version]]
  if (!downsamplings) {
    if (version == "v2") {
      base_fname <- glue("gnomad.v{full_version}.lof_metrics.by_gene.txt.bgz")
    } else if (version == "v4") {
      base_fname <- glue("gnomad.v{full_version}.constraint_metrics.tsv")
    }
  } else if (version == "v2") {
    base_fname <- "gnomad.v2.1.1.lof_metrics.downsamplings.txt.bgz"
  } else if (version == "v4") {
    base_fname <- glue("gnomad.v{full_version}.downsampling_constraint_metrics.tsv.bgz")
  }

  df <- read.delim(
  get_or_download_file(
    base_fname,
    version,
    output_basedir,
    data_subdir,
    release = release,
    public = public
  )
  )
  df <- mutate(df, !!version := TRUE)

  if (!downsamplings && (version == "v2")) {
    df <- rename(df, constraint_flags = constraint_flag) %>%
      mutate(
        constraint_flags = if_else(
            .data$constraint_flags == "",
            "[]",
            .data$constraint_flags
        )
      )
  }
  if (version == "v4") {
    df <- df %>%
    rename(
    exp_syn = syn.exp,
    exp_mis = mis.exp,
    exp_lof = lof.exp,
    obs_syn = syn.obs,
    obs_mis = mis.obs,
    obs_lof = lof.obs
  )
  }
  for(col_name in c("canonical", "mane_select")) {
    if (col_name %in% colnames(df)) {
      df <- mutate(df, !!col_name := .data[[col_name]] == "true")
    }
  }

  return(df)
}
```

There could be other things that you can add to this to keep these in loading and make the datasets consistent